### PR TITLE
Use Eclipse Temurin, not AdoptOpenJDK in publish artifact action

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Use Eclipse Temurin, not AdoptOpenJDK in publish artifact action

The 'adopt' distribution has moved to Eclipse Temurin and won't be updated at the AdoptOpenJDK location.

The [advanced usage](https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#adopt) describes the transition.

The [setup-java usage](https://github.com/actions/setup-java#usage) provides more general purpose info.

The current Adopt version is reporting:

```text
Run actions/setup-java@v2
Resolved Java 8.0.292+1 from tool-cache
Setting Java 8.0.292+1 as the default

Java configuration:
  Distribution: adopt
  Version: 8.0.292+1
  Path: /opt/hostedtoolcache/Java_Adopt_jdk/8.0.292-1/x64
```

The current Temurin version reports:

```text
Run actions/setup-java@v3.0.0
Resolved Java 8.0.322+6 from tool-cache
Setting Java 8.0.322+6 as the default

Java configuration:
  Distribution: temurin
  Version: 8.0.322+6
  Path: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/8.0.322-6/x64
```

The upgrade from JDK 8u292 to 8u322 is a positive step, since it is using a JDK version that is 12 months newer.

### Proposed changelog entries

N/A - skip changelog because there are no user visible results from this change

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

Any

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Review by StefanSpieker — APPROVED** [review]

File: .github/workflows/publish-release-artifact.yml
Comment (jglick) [inline .github/workflows/publish-release-artifact.yml:16]: https://github.com/jenkinsci/file-parameters-plugin/pull/90#discussion_r835918340 I _think_ also applicable here given https://github.com/jenkinsci/jenkins/blob/abd028bbb373a87708d60ad3d2e5b18b589c7e54/.github/workflows/publish-release-artifact.yml#L3-L5

Comment (basil) [reply .github/workflows/publish-release-artifact.yml:16]: What is the purpose of reviewing a PR that has already been merged?

Comment (jglick) [reply .github/workflows/publish-release-artifact.yml:16]: Just informational, in case the author (or anyone else) thinks it would be appropriate to file a follow-up.

(In this case I did not see this PR until after it is was merged, due to a link from a plugin PR.)

Comment (basil) [reply .github/workflows/publish-release-artifact.yml:16]: Have you considered doing a quick glance at the PRs with the `ready-for-merge` label every day or two to see if you have feedback for them? If the purpose of code review is to make the code of tangibly higher quality, then leaving feedback before a pull request is merged is far more likely to achieve this outcome. To be clear, there are many valid reasons for reviewing a PR after it has been merged, such as to notify authors of serious regressions, but in cases like this where there is no regression and where the suggestion is fairly trivial, it can come across as an example of "bring me a rock" style leadership.

Comment (jglick) [reply .github/workflows/publish-release-artifact.yml:16]: I am not familiar with that metaphor.

I do not normally attempt to follow all core PRs (or PRs in major plugins for that matter) since I simply lack the time for that sort of volume and there are already a number of diligent reviewers. This one I just stumbled across after the fact, via a link from a PR in a plugin I help maintain.

**Pull request merged**
